src/tribler/core/utilities/process_manager/process.py

```diff
@@ -177,8 +180,29 @@ def is_running(self):
             # The same PID value was reused for a new process, so the previous process is not running anymore
             return False
 
+        # Check if Core API is running if port is known
+        if self.kind == ProcessKind.Core and self.api_port is not None:
+            return self.is_api_running()
+
         return True
 
+    def is_api_running(self) -> bool:
+        """
+        Checks if REST API is running at the API port.
+
+        This is done by checking the Docs endpoint which is always available
+        and does not require API_KEY to access.
+        """
+        if not self.api_port:
+            return False
+
+        try:
+            docs_url = f"http://localhost:{self.api_port}/docs"
+            _ = requests.get(docs_url, timeout=API_CHECK_TIMEOUT_IN_SECONDS)
```

Comment:
The second fix proposed in this PR is an HTTP request to the Tribler REST API added to the `TriblerProcess.is_running` method.

Note that the `is_running` method is called, for example, when `TriblerProcess.__str__()` is called. It looks a bit extreme to me to issue an HTTP request whenever someone tries to print an instance of a core process.

Second, it is not clear why it is helpful to have this check for the Core process for two reasons:
1. If the logic for GUI processes is correct, the second GUI instance should understand that it is secondary and should not start the second Core at all. So, the check for the Core is less important than for the GUI if the GUI check is implemented correctly.
2. It is hard to say what this check allows us to achieve. Initially, when the Core process is just started, its `api_port` is always None. Not that with the current implementation, the GUI process does not pass the API port number to the Core process. The Core process opens the REST API on any available port and sets its value to the database so the GUI process can look up the port value. So when the Core process is just starting, its `api_port` is always None, and this check will not be triggered. When two Core processes are already running by mistake, their `api_port` values will differ. Each Core process has a different value of its API port and can connect to it (by the way, can the code create a deadlock when the Core process connects to its port? The call to `request.get` is synchronous, so it looks like it should block the asyncio loop). It is hard to say why this check of the HTTP endpoint availability is useful and how correctly it works in all different use cases.